src/test/kotlin/study/StringCalculator.kt

```diff
@@ -0,0 +1,72 @@
+package study
+
+class StringCalculator {
+    fun calculate(arithmeticExpression: String): Float {
+        if (arithmeticExpression.isBlank()) {
+            throw IllegalArgumentException("문자열이 비어있습니다.")
+        }
+
+        val splitedExpression = splitByBlank(arithmeticExpression)
+        val numbers = splitedExpression.filter { isNumber(it) }
+        val arithmeticSymbols = splitedExpression.filter { isOperator(it) }
+
+        validate(splitedExpression, numbers, arithmeticSymbols)
+
+        val result = numbers.map { it.toFloat() }
+            .toTypedArray()
+            .reduceIndexed { index, acc, s ->
+                doCalculate(acc, s, arithmeticSymbols[index - 1])
+            }
```

Comment:
코틀린은 컬렉션에 매우 많은 메서드를 제공하기에 `toTypedArray()` 없이도 `reduce`가 가능합니다!
```suggestion
        val result = numbers.map { it.toFloat() }
            .reduceIndexed {index, acc, s ->
                doCalculate(acc, s, arithmeticSymbols[index - 1])
            }
```

Reply:
`reduce` 💯